agent/rpcclient/health/health.go

```diff
@@ -81,8 +81,16 @@ func (c *Client) Notify(
 	ch chan<- cache.UpdateEvent,
 ) error {
 	cacheName := c.CacheName
-	if req.Ingress {
-		cacheName = c.CacheNameIngress
+	if req.Ingress || req.Source.Node != "" {
+		cacheName = c.CacheNameNotStreaming
 	}
 	return c.Cache.Notify(ctx, cacheName, &req, correlationID, ch)
 }
+
+func (c *Client) UseStreaming(req structs.ServiceSpecificRequest) bool {
+	if req.Ingress || req.Source.Node != "" {
+		return false
+	}
+
+	return req.QueryOptions.UseCache || req.QueryOptions.MinQueryIndex > 0
```

Comment:
why is `args.QueryOptions.UseCache` here? It wasn't here in the pre-PR equivalent, and the place it was present is used differently when evaluating the chain of boolean expressions.